lib/contentful/migration.ts

```diff
@@ -3,14 +3,29 @@ import { runMigration } from "contentful-migration"
 import { info, success } from "../logger"
 import { MigrationOptions, PendingMigration } from "../types"
 
+export type MigrationResult = {
+  successful: boolean
+  fileName: string
+}
+
 export async function runMigrations(
   pendingMigrations: PendingMigration[],
   options: MigrationOptions
-) {
+): Promise<MigrationResult[]> {
   const yes = options.yes !== undefined ? options.yes : true
-  for (const { filePath } of pendingMigrations) {
+  const migrationResult: MigrationResult[] = []
+  for (const { filePath, fileName } of pendingMigrations) {
     info(`Deploying migration ${filePath}...`)
-    await runMigration({ ...options, filePath, yes })
-    success(`${filePath} migration deployed`)
+
+    // disable this es-lint warning because the underlying function return "any" type and we have no control over that.
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const result = await runMigration({ ...options, filePath, yes })
+    if (result) {
+      success(`${filePath} migration deployed`)
+      migrationResult.push({ successful: true, fileName })
+    }
+    migrationResult.push({ successful: false, fileName })
```

Comment:
Maybe it's not a big deal since we only use the successful migrated files [here](https://github.com/foobaragency/cf-migrations/pull/87/files#diff-275a8ac6277615729852e43bbba224a48152d827d2503c3c2fba58eda9581b6eR38).

But the resulting array here is having duplicated results for the same file when the result is successful. For example, I tested with the following migrations:

<img width="286" alt="Screenshot 2021-07-05 at 12 45 50" src="https://user-images.githubusercontent.com/8534191/124459904-faef1880-dd8e-11eb-9499-7a913a86d5cf.png">

And I logged the `migrationResult` array and got the following:
<img width="546" alt="Screenshot 2021-07-05 at 12 46 37" src="https://user-images.githubusercontent.com/8534191/124459976-13f7c980-dd8f-11eb-9eab-155772319a29.png">

Obs: I skipped the migrations 0002 and 0003.

Do you think it could be problems like unexpected behaviours in the future if we add new futures that uses this result?

Reply:
Hi @marquesVF Thanks for the observation 🙈 . I fixed that with the recent commit. 

> Do you think it could lead to problems like unexpected behaviours in the future if we add new futures that use this result?

I don't understand exactly what you mean but I don't see any unexpected behaviours if we take into consideration that some migrations were not applied (skipped). since we save them in contentful, we can compare the local filesystem with the saved migration history.